### PR TITLE
Use the SageMaker Python SDK for local MNIST integration tests

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*
Corresponds to https://github.com/aws/sagemaker-pytorch-container/pull/87 and https://github.com/aws/sagemaker-pytorch-container/pull/88 (but for Chainer)

I left a TODO note about the way our MPI command construction doesn't preserve JSON encoding rather than fixing it to keep this PR small and narrowly-focused, but will hopefully fix it in a future PR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integration/local/test_mnist.py [hunk 1]
```diff
@@ -12,113 +12,161 @@
...
+    predictor.serializer = json_serializer
+    predictor.accept = 'application/json'
+    predictor.deserializer = json_deserializer
+    return predictor
 
-    assert not local_mode.file_exists(opt_ml, 'output/failure'), 'Failure happened'
 
-    with local_mode.serve(os.path.join(mnist_path, customer_script), model_dir=None,
-                          image_name=docker_image, opt_ml=opt_ml):
+class CsvDeserializer():
```
Comment (icywang86rui) [inline test/integration/local/test_mnist.py]: Can we add this to the python sdk?

File: test/resources/mnist/distributed_customer_script_with_env_vars.py [hunk 2]
```diff
@@ -162,8 +162,13 @@ def _preprocess_mnist(raw, withlabel, ndim, scale, image_dtype, label_dtype, rgb
 
     trainer.run()
 
+    # TODO: the MPI command makes the SM_HOSTS variable an invalid JSON
```
Comment (icywang86rui) [inline test/resources/mnist/distributed_customer_script_with_env_vars.py]: This is not really actionable. It's just a statement.

File: test/integration/local/test_mnist.py (hunk 1 above)
Comment (laurenyu) [reply test/integration/local/test_mnist.py]: sure - https://github.com/aws/sagemaker-python-sdk/pull/737

File: test/resources/mnist/distributed_customer_script_with_env_vars.py (hunk 2 above)
Comment (laurenyu) [reply test/resources/mnist/distributed_customer_script_with_env_vars.py]: will fix

File: test/utils/test_utils.py [hunk 3]
```diff
@@ -12,14 +12,51 @@
 # language governing permissions and limitations under the License.
 from __future__ import absolute_import
 
-import test.utils.local_mode as localmode
```
Comment (icywang86rui) [inline test/utils/test_utils.py]: Is local_mode.py still used? If not let's delete that.

Comment (laurenyu) [reply test/utils/test_utils.py]: yeah, it should still be used by the other test files. I'll delete it with the last PR in this series of migrating the tests.

**Pull request merged**